samples/extensions/vertex_dynamic_state/vertex_dynamic_state.cpp

```diff
@@ -0,0 +1,634 @@
+	                                 4, 7, 3,
+	                                 0, 3, 2,
+	                                 0, 2, 1,
+	                                 1, 2, 6,
+	                                 6, 5, 1,
+	                                 5, 6, 7,
+	                                 7, 4, 5,
+	                                 0, 1, 5,
+	                                 5, 4, 0,
+	                                 3, 7, 6,
+	                                 6, 2, 3};
+
+	struct
+	{
+		VkBuffer       buffer;
+		VkDeviceMemory memory;
+	} vertex_staging, index_staging;
+
+	vertex_staging.buffer = get_device().create_buffer(
+	    VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
+	    VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT,
+	    vertex_buffer_size,
+	    &vertex_staging.memory,
+	    vertices);
+
+	index_staging.buffer = get_device().create_buffer(
+	    VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
+	    VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT,
+	    index_buffer_size,
+	    &index_staging.memory,
+	    &indices);
+
+	cube.vertices = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                    vertex_buffer_size,
+	                                                    VK_BUFFER_USAGE_VERTEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                    VMA_MEMORY_USAGE_GPU_ONLY);
+
+	cube.indices = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                   index_buffer_size,
+	                                                   VK_BUFFER_USAGE_INDEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                   VMA_MEMORY_USAGE_GPU_ONLY);
+
+	/* Copy from staging buffers */
+	VkCommandBuffer copy_command = device->create_command_buffer(VK_COMMAND_BUFFER_LEVEL_PRIMARY, true);
+
+	VkBufferCopy copy_region = {};
+
+	copy_region.size = vertex_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    vertex_staging.buffer,
+	    cube.vertices->get_handle(),
+	    1,
+	    &copy_region);
+
+	copy_region.size = index_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    index_staging.buffer,
+	    cube.indices->get_handle(),
+	    1,
+	    &copy_region);
+
+	device->flush_command_buffer(copy_command, queue, true);
+
+	vkDestroyBuffer(get_device().get_handle(), vertex_staging.buffer, nullptr);
+	vkFreeMemory(get_device().get_handle(), vertex_staging.memory, nullptr);
+	vkDestroyBuffer(get_device().get_handle(), index_staging.buffer, nullptr);
+	vkFreeMemory(get_device().get_handle(), index_staging.memory, nullptr);
+
+	delete[] vertices;
+}
+
+/**
+ * 	@fn void vertex_dynamic_state::change_vertex_input_data(vertexDynamicStateVertexStruct_t variant)
+ * 	@brief Change Vertex Input data structure dynamically in runtime.
+ */
+void vertex_dynamic_state::change_vertex_input_data(vertexDynamicStateVertexStruct_t variant)
```

Comment:
If you reduce this function to just set the values that change (`vertex_bindings_description_ext.stride` and `vertex_attribute_description_ext[1].offset`), it would be more obvious what happens here.
Moreover, as there are just those two values that change, you could possibly completely remove this function (and that `vertexDynamicStateVertexStruct_t`) and set those values directly in `build_command_buffers()`.

Reply:
I agree with reducing elements that are inside the function. About removing whole function and hardcode these lines into `build_command_buffers` I have mixed feelings. The function name is clearly telling users what's going on before drawing models. I can implement your suggestion and for visibility, I will add more comments near hardcoded lines with a simple explanation.

Reply:
You're right, function name and argument are telling the right story.
But it essentially "hides" important work on two data structures behind a function call. And the uninformed reader has to realize, that the data that is used in `vkCmdSetVertexInputEXT` is modified right there. He has to jump back and forth in code to see the connection.
When the function calls would be replaced by just two lines of code each (maybe with some nice comment), the important parts here are in one place.

Reply:
I catch it and I agree with you. I will change it in the next commit.